### PR TITLE
Add support for tagging blocks as a keyword list

### PR DESCRIPTION
**Pull request by mononym**
Pretty straightforward change. Let me know if you have any questions.

### CODE REVIEW

File: lib/yaml_elixir/mapper.ex [hunk 1]
```diff
@@ -19,6 +19,13 @@ defmodule YamlElixir.Mapper do
   defp _to_map({:yamerl_map, :yamerl_node_map, _tag, _loc, map_tuples}, options),
     do: _tuples_to_map(map_tuples, empty_container(options), options)
 
+  defp _to_map({:yaml_elixir_keyword_list, _module, _tag, _loc, tuples}, options) do
+    case _tuples_to_map(tuples, empty_container(options), options) do
+      map = %{} -> for {key, value} <- map, do: {key, value}
```
Comment (KamilLelonek) [inline lib/yaml_elixir/mapper.ex]: I suggest to avoid `case` here:

```elixir
defp _to_map({:yaml_elixir_keyword_list, _module, _tag, _loc, tuples}, options) do
  tuples
  |> _tuples_to_map(empty_container(options), options)
  |> to_keyword_list()
end
```

and then define a proper `to_keyword_list/1` with guards like `when is_map(map)` for example.

File: test/yaml_elixir_test.exs [hunk 2]
```diff
@@ -214,6 +214,14 @@ defmodule YamlElixirTest do
...
   end
 
+  test "should receive maps as keyword lists when map is tagged in the file" do
+    assert_parse_file(
+      "keyword_list",
+      %{"prod" => %{"foo" => [{"bar", "foo"}, {"foo", "bar"}]}},
+      node_mods: [YamlElixir.Node.KeywordList]
+    )
+  end
+
```
Comment (KamilLelonek) [inline test/yaml_elixir_test.exs:224]: Can you add more test examples to verify different and edge cases?

File: lib/yaml_elixir/yaml_node_keyword_list.ex [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+    error =
+      yamerl_parsing_error(
+        name: :not_a_mapping,
+        token: token,
+        text: "Invalid mapping",
+        line: elem(unfinished_node(node, :pres), 0),
+        column: elem(unfinished_node(node, :pres), 1)
+      )
+
+      throw(error)
```
Comment (KamilLelonek) [inline lib/yaml_elixir/yaml_node_keyword_list.ex:87]: Why `throw` here instead of `raise`?

File: lib/yaml_elixir/yaml_node_keyword_list.ex [hunk 4]
```diff
@@ -0,0 +1,95 @@
+defmodule YamlElixir.Node.KeywordList do
+  require Record
+  import YamlElixir.Records, only: :macros
+
+  @tag 'tag:yaml_elixir,2019:keyword_list'
+
+  def tags() do
+    [@tag]
+  end
```
Comment (KamilLelonek) [inline lib/yaml_elixir/yaml_node_keyword_list.ex]: You can do it more elegant:

```elixir
def tags, do: [@tag]
```

File: lib/yaml_elixir/yaml_node_keyword_list.ex (hunk 3 above)
Comment (mononym) [reply lib/yaml_elixir/yaml_node_keyword_list.ex:87]: It's what they do in yamerl: https://github.com/yakaz/yamerl/blob/master/src/yamerl_node_map.erl#L115

And since I'm implementing another node module I wanted to follow the conventions they set.

File: test/yaml_elixir_test.exs (hunk 2 above)
Comment (mononym) [reply test/yaml_elixir_test.exs:224]: What edge cases are you thinking? Aside from having a yaml doc with a single tagged value, rather than tagging a value in a nested map, what scenarios do you have in mind?

File: lib/yaml_elixir/mapper.ex (hunk 1 above)
Comment (mononym) [reply lib/yaml_elixir/mapper.ex]: Done

File: lib/yaml_elixir/yaml_node_keyword_list.ex (hunk 4 above)
Comment (mononym) [reply lib/yaml_elixir/yaml_node_keyword_list.ex]: Done

File: test/yaml_elixir_test.exs (hunk 2 above)
Comment (KamilLelonek) [reply test/yaml_elixir_test.exs:224]: In the issue you posted previously, there were more than one case explained. I'm wondering whether there are more examples you can put in the tests.

Comment (mononym) [reply test/yaml_elixir_test.exs:224]: One thing I've done is modify the existing test to have this yaml doc
~~~~
prod:
  foo: !<tag:yaml_elixir,2019:keyword_list>
    foo: bar
    bar: foo
  bar:
    foo: bar
    bar: foo
~~~~

To show that it will turn the tagged value into a keyword list while retaining the untagged value as a map

Comment (KamilLelonek) [reply test/yaml_elixir_test.exs:224]: OK

**Comment by mononym** [comment]
Added an additional test to show that a parsing error is expected when trying to tag a flow style map

**Review by KamilLelonek — APPROVED** [review]
